jsapp/js/components/modalForms/formMedia.es6

```diff
@@ -0,0 +1,153 @@
+import React from 'react';
+import autoBind from 'react-autobind';
+import alertify from 'alertifyjs';
+import Dropzone from 'react-dropzone';
+import {bem} from 'js/bem';
+import {dataInterface} from 'js/dataInterface';
+import {
+  t,
+} from 'js/utils';
+import {
+} from 'js/constants';
+
+/*
+Modal for uploading form media, part of #2487.
+*/
+class FormMedia extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      uploadedAssets: null
+    };
+
+    autoBind(this);
+  }
+
+  /*
+   * setup
+   */
+
+  componentDidMount() {
+    dataInterface.getFormMedia(this.props.asset.uid).done((uploadedAssets) => {
+      this.setState({uploadedAssets: uploadedAssets.results});
+    });
```

Comment:
I see this is still WIP, but you could encapsulate this call in a "refreshFormMedia" function, as you will call it multiple times.